components/post_view/combined_system_message/combined_system_message.jsx

```diff
@@ -313,17 +323,23 @@ export default class CombinedSystemMessage extends React.PureComponent {
                     userIds = [this.props.currentUserId];
                 } else {
                     // Not something the current user did or was affected by
-                    continue;
+                    return;
                 }
             }
 
-            content.push(
-                <React.Fragment key={postType + actorId}>
-                    {this.renderFormattedMessage(postType, userIds, actorId)}
-                    <br/>
-                </React.Fragment>
-            );
-        }
+            if (postType === REMOVE_FROM_CHANNEL) {
+                removedUserIds = removedUserIds.concat(userIds);
+            } else if (postType !== REMOVE_FROM_CHANNEL && removedUserIds.length > 0) {
+                content.push(this.renderMessage(postType, removedUserIds, actorId));
```

Comment:
Why switch how the "User was removed" messages are combined? Is this because they're separated by actorId now internally, but we still want to render them combined?

Reply:
Yes, that's right.  If in the future we wanted to show it with `...removed by {user}`, then we'll not need this logic.  I didn't ask to do it now since it's better to have it at the same time with `remove from team`.  Unfortunately that `remove from team` has different props implementation and so it will require server changes and/or require backwards compatibility which is a bit work to include here.

Reply:
Hmmm, I realized you're right with your loop comment on mobile RN.  I'll update this loop logic.

Reply:
Thanks @hmhealey,  I've updated the logic.  Also I submitted a PR (https://github.com/mattermost/mattermost-redux/pull/586) to mattermost-redux but it's not blocking to how this change works.